clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchDocumentMapper.scala

```diff
@@ -1,29 +1,50 @@
 package org.broadinstitute.clio.server.dataaccess.elasticsearch
 
+import io.circe.{Encoder, Json}
+import org.broadinstitute.clio.util.generic.CaseClassMapper
+import org.broadinstitute.clio.util.model.{EntityId, UpsertId}
+
+import scala.reflect.ClassTag
+
 /**
   * Maps metadata to an Elasticsearch document.
   *
-  * @tparam ModelKey      The type of key.
-  * @tparam ModelMetadata The type of the metadata.
-  * @tparam Document      The type of the document.
+  * @tparam Key      The type of the TransferKey.
+  * @tparam Metadata The type of the TransferMetadata.
   */
-abstract class ElasticsearchDocumentMapper[ModelKey, ModelMetadata, Document] {
+class ElasticsearchDocumentMapper[Key <: Product: ClassTag, Metadata: ClassTag] (genId: () => UpsertId)(
+  implicit
+  private[elasticsearch] val keyEncoder: Encoder[Key],
+  private[elasticsearch] val metadataEncoder: Encoder[Metadata]
```

Comment:
Since these aren't used explicitly, then can be moved up into the type signature as context bounds. i.e.:
`Key <: Product: Encoder`